src/components/Auth/useAuthProvider.ts

```diff
@@ -1,54 +1,34 @@
-import { useMutation, UseMutationOptions } from '@tanstack/react-query'
 import { useClient } from '@vocdoni/react-providers'
 import { RemoteSigner } from '@vocdoni/sdk'
-import { useCallback, useMemo, useState } from 'react'
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import { useMutation, UseMutationOptions } from '@tanstack/react-query'
+import { api, ApiEndpoints, ApiParams, UnauthorizedApiError } from '~components/Auth/api'
 
-type LoginResponse = { token: string; expirity: string }
+export type LoginResponse = { token: string; expirity: string }
 
-export interface ILoginParameters {
+export interface ILoginParams {
   email: string
   password: string
 }
 
-export interface IRegisterParameters {
+export interface IRegisterParams {
   firstName: string
   lastName: string
   email: string
   password: string
 }
 
-type MethodTypes = 'GET' | 'POST' | 'PUT' | 'DELETE'
-
-const api = <T>(path: string, params?: unknown, method: MethodTypes = 'GET'): Promise<T> => {
-  return fetch(`${import.meta.env.SAAS_URL}${path}`, {
-    method,
-    headers: {
-      'Content-Type': 'application/json',
-    },
-    body: JSON.stringify(params),
-  })
-    .then((response) => {
-      if (!response.ok) {
-        throw new Error(response.statusText)
-      }
-      return response.json() as Promise<T>
-    })
-    .catch((error: Error) => {
-      // todo(kon): implement error handling
-      throw error
-    })
-}
-
-const useLogin = (options: Omit<UseMutationOptions<LoginResponse, Error, ILoginParameters>, 'mutationFn'>) => {
-  return useMutation<LoginResponse, Error, ILoginParameters>({
-    mutationFn: (params: ILoginParameters) => api<LoginResponse>('auth/login', params, 'POST'),
+export const useLogin = (options: Omit<UseMutationOptions<LoginResponse, Error, ILoginParams>, 'mutationFn'>) => {
```

Comment:
@selankon Considering that there are going to be more queries needed in the future, I  think it makes sense to have the queries implmented in the query folder. What do you think?

Reply:
Same as on the

Reply:
8060499